functions/src/plugins/rerun-circleci.ts

```diff
@@ -98,4 +97,73 @@ ${error.message}
     const repositoryConfig = await this.getAppConfig(context);
     return repositoryConfig.rerunCircleCI;
   }
+
+  /**
+   * Get the CircleCI workflow id of the first discovered CircleCI status in the statuses.  Since
+   * only one workflow is run on each PR, all CircleCI statuses will track back to the same
+   * workflow id.
+   */
+  private async getCircleCiWorkflowIdForPullRequest(context: Context) {
+    /** The target url of the discovered CircleCI status. */
+    let targetUrl: string;
+    /** The pull request which triggered the bot action. */
+    const pullRequest: Github.PullRequestsGetResponse = context.payload.pull_request;
+    /** The owner and repository name. */
+    const {owner, repo} = context.repo();
+    /** The list of statuses for the latest ref of the PR. */
+    const {statuses} = (await context.github.repos.getCombinedStatusForRef({
+      owner, repo, ref: pullRequest.head.ref
+    })).data;
+
+    for (const status of statuses) {
+      if (status.context.startsWith('ci/circleci:')) {
+        targetUrl = status.target_url;
+        break;
+      }
+    }
```

Comment:
Super-nit: This could be simplified as:

```suggestion
    const targetUrl = statuses.find(x => x.context.startsWith('ci/circleci:'))?.target_url;
```